### PR TITLE
Minimalistic

### PR DESCRIPTION
**Pull request by vsemp**
No to merge. Only for review.

### CODE REVIEW

**Comment by portante** [comment]
@vsemp, why would you not want to merge this?

**Comment by portante** [comment]
@vsemp, we should update the .gitignore file so that the Mac's `.DS_Store` files are not included in commits.

File: bin/quads.py [hunk 1]
```diff
@@ -276,9 +276,9 @@ def main(argv):
         exit(0)
 
     if args.movehosts:
-        if args.datearg is not None and not args.dryrun:
-            print "--move-hosts and --date are mutually exclusive unless using --dry-run."
-            exit(1)
+    #    if args.datearg is not None and not args.dryrun:
```
Comment (portante) [inline bin/quads.py:279]: Why comment this out?

File: bin/move-and-rebuild-host-hil.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    # for nic_json in node_nics:
+    #     node_detach_network(node_name,
+    #                         nic_json['label'],
+    #                         old_cloud)
+    #     node_connect_network(node_name,
+    #                          nic_json['label'],
+    #                          new_cloud)
+
+if __name__ == "__main__":
+   move_host(sys.argv)
```
Comment (portante) [inline bin/move-and-rebuild-host-hil.py:22]: Missing trailing new line.

File: bin/move-and-rebuild-host-hil.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/python
+
+import sys
+import cli
+
+def move_host(argv):
+    host_to_move = argv[1]
+    old_cloud = argv[2]
+    new_cloud = argv[3]
+    print(show_node(host_to_move))
```
Comment (portante) [inline bin/move-and-rebuild-host-hil.py:10]: This appears to do nothing.  Why is that?

File: bin/move-and-rebuild-host-hil.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
```
Comment (portante) [inline bin/move-and-rebuild-host-hil.sh:1]: Is this a copy of an existing script but with the "move hosts" part invoked as a python script?  Could you add a comment to that effect, if so, otherwise, let's try to explain in a comment what the purpose of this script is.

File: hil_extension/README.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+HIL folder and QUADS folder should be in the same parent directory.
```
Comment (portante) [inline hil_extension/README.md:1]: Do you mean folders?  Or checkout git repos of HIL and QUADS?

File: hil_extension/err.txt [hunk 6]
```diff
@@ -0,0 +1,55 @@
+INFO:werkzeug: * Running on http://127.0.0.1:5000/
```
Comment (portante) [inline hil_extension/err.txt:1]: What is `err.txt` used for?

File: hil_extension/cli.py [hunk 7]
```diff
@@ -0,0 +1,875 @@
+#!/bin/python
+
+# Copyright 2013-2014 Massachusetts Open Cloud Contributors
```
Comment (portante) [inline hil_extension/cli.py:3]: Is this copied from the HIL repo?  If so, why can't we just reference from there?  If it isn't, why do we have a copyright license from the MOC?

File: hil_extension/hil.yml [hunk 8]
```diff
@@ -0,0 +1 @@
+url: http://127.0.0.1:5000
```
Comment (portante) [inline hil_extension/hil.yml:1]: What is `hil.yml` for?

File: hil_extension/load-config.sh [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# For existing shell scripts to use the ../conf/quads.yml
```
Comment (portante) [inline hil_extension/load-config.sh:1]: Can you comment as to why this has to be `source`d? And why it can't be just included in its parent script?

File: hil_extension/load-config.sh [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+    if [ ! -e $(dirname $0)/../conf/quads.yml ]; then
+        echo "$(basename $0): could not find $(dirname $0)/../conf/quads.yml"
+        exit 1
+    fi
+
+    yay $(dirname $0)/../conf/quads.yml
+}
+
+load_data
```
Comment (portante) [inline hil_extension/load-config.sh:21]: Why define a function only to execute it immediately?

File: bin/move-and-rebuild-host-hil.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
+#!/bin/python
```
Comment (portante) [inline bin/move-and-rebuild-host-hil.py:1]: Why is this file added and the other file in `hil_extension` of the same name?

File: hil_extension/yay.sh [hunk 12]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+# yay lets you parse config values in shells scripts from
+# our global ../conf/quads.yml configuration.  This lets us
+# have one authoritative place for configuration across all
+# tooling within QUADS.  Python has a library for this already
+# so this is only needed for shell and expect.
+# Original credit goes to:
+# https://github.com/HumanNeuroscienceLab/gunther/blob/master/include/yay.sh
```
Comment (portante) [inline hil_extension/yay.sh:8]: Why not just use that `yay.sh` script directly?  I don't think we can add this to the QUADS repo without proper attribution, and it seems easy enough to just fetch that one and use it.  Any reason why not?

**Review by portante — CHANGES_REQUESTED** [review]

File: hil_extension/yay.sh (hunk 12 above)
Comment (vsemp) [reply hil_extension/yay.sh:8]: Actually, yay.sh is already used by Red-Hat. It is in quads/bin

I made another copy in quads/lib to make my code completely independent of the rest Quads

File: hil_extension/load-config.sh (hunk 9 above)
Comment (vsemp) [reply hil_extension/load-config.sh:1]: Actually, load-config.sh is already used by Red-Hat. It is in quads/bin. It was added by people from Red-Hat.

I made another copy in quads/lib to make my code completely independent of the rest Quads

File: hil_extension/hil.yml (hunk 8 above)
Comment (vsemp) [reply hil_extension/hil.yml:1]: It's to keep information about the HIL server. We connect to this address. And we need to know this address upfront.

File: hil_extension/cli.py (hunk 7 above)
Comment (vsemp) [reply hil_extension/cli.py:3]: This is indeed copied. But I completely rewrote this code already. Unfortunately, my rewritten version is not in this PR.

File: hil_extension/err.txt (hunk 6 above)
Comment (vsemp) [reply hil_extension/err.txt:1]: It's just the output logs.  

./test.sh > log.txt 2> err.txt

File: hil_extension/README.md (hunk 5 above)
Comment (vsemp) [reply hil_extension/README.md:1]: I meant folders. 

I also rewrote this script already.

File: bin/move-and-rebuild-host-hil.sh (hunk 4 above)
Comment (vsemp) [reply bin/move-and-rebuild-host-hil.sh:1]: Completely agree.

The code (my code) in general is ugly and bad (but working). Orran said I should stop working on this minimalistic version and join the others because that's the aim of the class.

File: bin/move-and-rebuild-host-hil.py (hunk 3 above)
Comment (vsemp) [reply bin/move-and-rebuild-host-hil.py:10]: Yes. It's for debugging.

File: bin/quads.py (hunk 1 above)
Comment (vsemp) [reply bin/quads.py:279]: Because we want to use both flags for testing.

**Comment by saacg** [comment]
closing as it has been replaced by newer changes
